src/mapping/Mapping.cpp

```diff
@@ -44,6 +46,32 @@ Mapping::Constraint Mapping::getConstraint() const
   return _constraint;
 }
 
+bool Mapping::isIterative() const
+{
+  return _mappingType == Type::Iterative;
+}
+
+bool Mapping::hasInitialGuess() const
+{
+  PRECICE_ASSERT(isIterative(), "This mapping isn't iterative.");
+  PRECICE_ASSERT(_initialGuess != nullptr, "The last solution wasn't provided.");
+  return _initialGuess->size() > 0;
+}
+
+const Eigen::VectorXd &Mapping::initialGuess() const
+{
+  PRECICE_ASSERT(isIterative(), "This mapping isn't iterative.");
+  PRECICE_ASSERT(_initialGuess != nullptr, "The last solution wasn't provided.");
+  return *_initialGuess;
+}
+
+Eigen::VectorXd &Mapping::initialGuess()
+{
+  PRECICE_ASSERT(isIterative(), "This mapping isn't iterative.");
+  PRECICE_ASSERT(_initialGuess != nullptr, "The last solution wasn't provided.");
+  return *_initialGuess;
+}
```

Comment:
This seems to be a duplicate of the `const` version. Maybe call the one from the other?

Reply:
I would really prefer not to rely on `const_cast` for such a simple function. :sweat_smile: